### PR TITLE
Story 31 - 33

### PR DESCRIPTION
**Pull request by jhgould**
FINALLY!!! Sorry this took so long.

### CODE REVIEW

File: app/controllers/admin/dashboard_controller.rb [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
 
     def show
-    end 
+      @admin = current_user
+      @users = User.all
+      @orders = Order.all
+      @packaged_orders = Order.packaged_orders
+      @pending_orders = Order.pending_orders
+      @shipped_orders = Order.shipped_orders
+      @cancelled_orders = Order.cancelled_orders
```
Comment (jhgould) [inline app/controllers/admin/dashboard_controller.rb:10]: I dont think creating all these variables was the best way but its the way I got it to work for now.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -17,4 +16,13 @@ def current_admin?
     current_user && current_user.admin?
   end
 
+  def admin?
+    role == 2
+  end
+
+  def merchant?
+    role == 1
+  end
```
Comment (jhgould) [inline app/controllers/application_controller.rb:25]: this is so I could check and see if the current user was either an admin or a merchant. The controller would then direct them accordingly.

File: app/controllers/orders_controller.rb [hunk 3]
```diff
@@ -30,6 +28,12 @@ def create
     end
   end
 
+  def ship
+    order = Order.find(params[:id])
+    order.update!(status: "shipped")
+    redirect_to "/admin/dashboard"
```
Comment (jhgould) [inline app/controllers/orders_controller.rb:34]: This 'ship' method is for an admin. When its clicked, the order is 'shipped' and then the user is no longer able to cancel it.

File: app/controllers/profile_orders_controller.rb [hunk 4]
```diff
@@ -5,9 +5,10 @@ def index
   end
 
   def show
-
     @order = Order.find(params[:id])
-
+    if @order.status == "cancelled" || @order.status == "shipped"
+    else
+      @order.update(status: @order.find_order_status(params[:id]))
```
Comment (jhgould) [inline app/controllers/profile_orders_controller.rb:11]: Should probably refactor but Eric and I got this to work so in the wise words of some redneck "if it works don't fix it"

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -5,6 +5,17 @@ def new
...
+    # if @user.save
+    #   session[:id] = @user.id
+    #   session[:user_id] = @user.id
+    #   flash[:notice] = "You are now logged in #{user_params[:name]}"
+    #   redirect_to "/profile"
+    # else
+    #   flash[:error] = @user.errors.full_messages.join(". ").to_s
+    #   redirect_to "/register"
+    # end
+
```
Comment (jhgould) [inline app/controllers/users_controller.rb:18]: My mentor and I went over this and found a better way. I wanted to leave it as is but had to change it to make some of the admin stuff work. Sorry for whoever wrote the code I had to change.

File: app/models/order.rb [hunk 6]
```diff
@@ -9,9 +9,44 @@ def total_quantity
...
+    where(status: "pending")
+  end
+
+  def self.shipped_orders
+    where(status: "shipped")
+  end
+
+  def self.cancelled_orders
+    where(status: "cancelled")
+  end
```
Comment (jhgould) [inline app/models/order.rb:49]: Created model methods to order and list stuff in the view.

File: app/views/admin/dashboard/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      <section id=cancelled>
+       <% @cancelled_orders.each do |order| %>
+          <ul>
+          <li><p1><%=link_to order.name, "/profile" %></p1></li>
+          <li><p1><%=order.id  %></p1></li>
+          <li><p1><%=order.created_at  %></p1></li>
+          </ul>
+        <% end %>
+      </section>
+  </section>
```
Comment (jhgould) [inline app/views/admin/dashboard/show.html.erb:43]: Behold the most OCD code I have ever written.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (EricLarson2020) [reply app/controllers/application_controller.rb:25]: for this the enums actually has a method called admin? and merchant? which will check based on the role provided

**Pull request merged**